scripts/java-analyze.sh

```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/env bash
+
+set -x # Show commands
+set -eu # Errors/undefined vars are fatal
+set -o pipefail # Check all commands in a pipeline
+
+if [ $# -ne 2 ]
+then
+    echo "Usage: java-analyze.sh config-file.json tree_name"
+    exit 1
+fi
+
+CONFIG_FILE=$(realpath $1)
+TREE_NAME=$2
+
+cd $MOZSEARCH_PATH/java
+bazel run //:JavaAnalyze -- $FILES_ROOT $INDEX_ROOT/analysis
+cd -
```

Comment:
It'd be great to have a comment here that details or cross-references other documentation (ex: the `java/README.md) that explains the expected parallelism here and any limitations on this.  For example, https://github.com/mozsearch/mozsearch/blob/master/scripts/js-analyze.sh which drives in the similarly happens-during-searchfox-indexing JS analysis stage uses `parallel` which makes it clear how we're achieving parallelism.

If we aren't able to achieve parallelism in this stage, this may be something where we need to move it into the per-tree searchfox taskcluster jobs which currently have a lot of slack timing-wise.  That is, right now every additional minute that this step takes is an additional minute of latency for searchfox, whereas we literally can add hours of additional processing into the taskcluster jobs and it won't affect latency because there's already a large wait there for the code coverage data.

Reply:
As an addendum to my prior comment, it looks like whatever is happening is more than sufficiently performant for our purposes at this time, but it would be great to have this info available here/somewhere.

Reply:
Java parser runs on multi-thread and resolving "type" uses multi-thread.  Even if using `parallel`, it doesn't become faster.

When using parallel, 168 secs on my workstation (Ryzen 9 3900), but when not using it, 75 secs.

Reply:
Excellent!  Thank you for the clarification.  That definitely explains how it's so fast! :)